clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3155,6 +3155,16 @@ implicitly inherit this attribute.
   }];
 }
 
+def SYCLGlobalVarDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "SYCL Global Var";
+  let Content = [{
+This attribute can only be applied to global variables and indicates that the
+variable can be used in a SYCL kernel. Access to global variable is expected to
+be only from device-side.
```

Comment:
```suggestion
This attribute can only be applied to a global variable and indicates that the
variable can be used in a SYCL kernel. Access to the global variable is expected to
be only from device-side.
```
It'd be helpful to nail down what is and isn't a global variable a bit better. e.g., what about these cases (which may make for good test cases):
```
struct S {
  static int i;
};
int S::i = 0; // Is S::i a global variable?

static struct T {
  int i;
} t; // Is t.i a global variable because t is?

static union {
  int i;
}; // Is i a global variable because the anonymous union is at file scope?
```
Also, the docs could use some code examples of correct usage.

What does it mean that the access is expected to only be from the device side?

Reply:
Latest commit expands test cases and docs.